src/scripts.js

```diff
@@ -274,35 +278,38 @@ function buildRecipeCard(recipe, recipeCard, abilityToCook) {
       }" src="${flagFavoritedRecipes(recipe)}" alt="star icon"/>
     </section>
   `;
+  if (cookbookFlag) {
+    updateRecipeCardButtons(recipe, recipeCard, abilityToCook);
+  }
+}
 
+function updateRecipeCardButtons(recipe, recipeCard, abilityToCook) {
   const cookRecipeContainer = document.createElement('div');
   cookRecipeContainer.classList.add('cook-recipe-container');
-
   const abilityToCookBtn = document.createElement('button');
   abilityToCookBtn.classList.add('recipe-card-button');
   abilityToCookBtn.setAttribute('id', 'recipeCardButton');
   abilityToCookBtn.innerText = `${abilityToCook}`;
-  abilityToCookBtn.addEventListener('click', function () {
-    displayCookRecipePopUp(event, recipe.id);
+  abilityToCookBtn.dataset.recipeId = `${recipe.id}`;
+  abilityToCookBtn.addEventListener('click', function (event) {
+    displayCookRecipePopUp(event, recipe);
```

Comment:
Created a `dataset id` for `abilityToCookBtn` so that the event handlers will have access to the `recipe ids` that they need.